mlflow/utils/search_utils.py

```diff
@@ -430,20 +491,212 @@ def _parse_order_by_string(cls, order_by):
     @classmethod
     def parse_order_by_for_search_runs(cls, order_by):
         token_value, is_ascending = cls._parse_order_by_string(order_by)
-        identifier = cls._get_identifier(token_value.strip(), cls.VALID_ORDER_BY_ATTRIBUTE_KEYS)
+        identifier = cls._get_identifier(
+            token_value.strip(), cls.VALID_ORDER_BY_RUN_ATTRIBUTE_KEYS, True
+        )
         return identifier["type"], identifier["key"], is_ascending
 
     @classmethod
     def parse_order_by_for_search_registered_models(cls, order_by):
         token_value, is_ascending = cls._parse_order_by_string(order_by)
-        token_value = token_value.strip()
-        if token_value not in cls.VALID_ORDER_BY_KEYS_REGISTERED_MODELS:
-            raise MlflowException(
-                "Invalid order by key '{}' specified. Valid keys ".format(token_value)
-                + "are '{}'".format(cls.RECOMMENDED_ORDER_BY_KEYS_REGISTERED_MODELS),
-                error_code=INVALID_PARAMETER_VALUE,
+        identifier = cls._get_identifier(
+            token_value.strip(), cls.VALID_ORDER_BY_REGISTERED_MODEL_ATTRIBUTE_KEYS, False,
+        )
+        # last updated timestamp field in search registered model has alias,
+        # we want to map those alias to the correct db key
+        if identifier["key"] in SearchUtils.VALID_TIMESTAMP_ORDER_BY_KEYS:
+            identifier["key"] = SqlRegisteredModel.last_updated_time.key
+        return identifier["type"], identifier["key"], is_ascending
+
+    @classmethod
+    def _get_subquery_entity(cls, key_type, comparator, is_search_run):
+        if is_search_run:
+            if cls.is_metric(key_type, comparator):  # any valid comparator
+                return SqlLatestMetric
+            elif cls.is_tag(key_type, comparator):
+                return SqlTag
+            elif cls.is_param(key_type, comparator):
+                return SqlParam
+            elif cls.is_attribute(key_type, comparator):
+                return None
+            else:
+                raise MlflowException(
+                    "Invalid identifier type '%s'" % key_type, error_code=INVALID_PARAMETER_VALUE,
+                )
+        else:
+            if cls.is_tag(key_type, comparator):
+                return SqlRegisteredModelTag
+            elif cls.is_attribute(key_type, comparator):
+                return None
+            else:
+                raise MlflowException(
+                    "Invalid identifier type '%s'" % key_type, error_code=INVALID_PARAMETER_VALUE,
+                )
+
+    @classmethod
+    def _get_order_by_clauses(cls, order_by_list, session, is_search_run):
```

Comment:
As with the other backend PR, I think we should try to avoid these boolean flag-based inputs / behavioral modifiers for utilities. Rather than providing a `_get_order_by_clauses` method that is parameterized by the `is_search_run` boolean, I'd rather we extract out common logic for order by construction into one or more utility methods and then invoke these utility methods from `get_order_by_clauses_for_run` and `get_order_by_clauses_for_registered_model`. This will make it easier to understand what's going on (right now, the control flow of this method is complex / hard to reason about for various input cases).

This same comment applies to other utility methods in this module that fork behavior based on `is_search_run`; if we have to, I think it's better to duplicate a bit of code than create aggregate routines with lots of branching logic / complex control flow.